api/admin/controller/__init__.py

```diff
@@ -2432,20 +2430,31 @@ def _get_prior_test_results(self, item, protocol_class=None):
                         self._db, protocol_class, self._db, item, *extra_args
                     )
 
-            else:
-                # We're running self-tests for a search service
-                item_type = "search service"
+            elif self.type == "search service":
                 self_test_results = ExternalSearchIndex.prior_test_results(
                     self._db, None, self._db, item
                 )
 
+            elif self.type == "patron authentication service":
+                library = None
+                if len(item.libraries):
+                    library = item.libraries[0]
+                    self_test_results = protocol_class.prior_test_results(
+                        self._db, None, library, item
+                    )
+                else:
+                    self_test_results = dict(
+                        exception=_("You must associate this service with at least one library before you can run self tests for it."),
+                        disabled=True
```

Comment:
Can you return a problem detail document instead of this custom data structure?

Reply:
I think so...but I'm pretty sure that (unless I special-case the front-end rendering in a weird, unintuitive way) that will make it display like an error message.  Is that what we want here?

Reply:
I see, this is being displayed in the context of the test results. I ran into a similar problem myself just now. Leave this alone for now and we can revisit improving the handling of unusual error cases later.